csharp/ql/test/library-tests/dataflow/flowsources/aspremote/AspRemoteFlowSource.cs

```diff
@@ -0,0 +1,20 @@
+using Microsoft.AspNetCore.Mvc;
+
+namespace Testing
+{
+
+    public class ViewModel
+    {
+        public string RequestId { get; set; }
+
+        public object Query;
+    }
```

Comment:
Nit picking, but what happens with `get`-only properties and `private` members?

Reply:
Hmm, they are also included. There is no further restrictions on the Assignable members. Let's change this such that it is only public and non static assignable members.
Good catch!

Reply:
Also - this is not nit picking! Really good that you ask these questions!

Reply:
We can also fine-grain it even more. Do you have any special opinion on, which members we should included/exclude?

Reply:
@tamasvajk : Thank you!
Ah yes, I think you are right. I read some of this (I know it is old): https://docs.microsoft.com/en-us/archive/msdn-magazine/2012/february/asp-net-mvc-the-features-and-foibles-of-asp-net-mvc-model-binding

It doesn't explicitly state that properties should be get/set, but it seems like it is implicitly assumed and this could serve is a conservative extension of the existing logic (cover more with lower risk of false positives).
My suggestion would be that we take all public properties with public getters and setters and all public non-readonly fields. No statics allowed of course. Then we should be on the safe side and also cover most cases!

Reply:
I think we can ignore fields altogether. There are some commonly used attributes to specify from where data should be bound. For example if data is taken from the request body, then `FromBody` is used: https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.mvc.frombodyattribute?view=aspnetcore-6.0. These attributes have no `Field` target, so I think we can safely ignore fields.

Reply:
Alright! ASP.NET is complicated :-) It looks like the canonical examples all uses public (non static) auto implemented properties with public getters and setters. Maybe that is sufficiently conservative as a starting point.